integration-tests/kotlin-serialization/src/test/kotlin/io/quarkus/it/kotser/ResourceTest.kt

```diff
@@ -0,0 +1,40 @@
+package io.quarkus.it.kotser
+
+import io.quarkus.test.junit.QuarkusTest
+import io.restassured.RestAssured.given
+import io.restassured.http.ContentType.JSON
+import org.hamcrest.CoreMatchers.`is`
+import org.junit.jupiter.api.Test
+
+@QuarkusTest
+class ResourceTest {
```

Comment:
Shouldn't there be more testing done here? I mean aren't there specific things we would like to know that work in Kotlin Serialization that for example don't work in Jackson?

Reply:
I open to ideas but we're sending and receiving objects that are being marshalled via kotlin serialization at this point. a lot of the configuration stuff is how kotser produces json and there's a test or two on the server side for that (checking that configuration is applied) but at the point we'd be testing that kotser is doing what it promised rather than testing quarkus.  There *is* one more test I'd like to add (recalling a failed test earlier that got removed once I reworked the tests.

Reply:
For Jackson, we have various tests for features enabled by config - see https://github.com/quarkusio/quarkus/tree/main/extensions/jackson/deployment/src/test/java/io/quarkus/jackson/deployment

Reply:
I am mainly thinking of how we test **our** integration with kotser, not testing every feature of kotser itself

Reply:
There's a test on the server side that checks that `prettyIndent` is applied, e.g., to ensure that the configuration values are being applied properly.  Is that what you're looking for?

Reply:
Yeah, but I would like to see at least one more

Reply:
But if you add the `deployment` tests (see other comment), we don't have to worry about this.